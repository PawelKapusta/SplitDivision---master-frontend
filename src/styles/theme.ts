import { DefaultTheme } from "styled-components";

export const theme: DefaultTheme = {
    palette: {
        white: "#fff",
        black: "#000",
        background1: "#0F1624",
        gold: "#eaaf57",
    },
    breakpoints: {
        sm: "screen and (max-width: 640px)",
        md: "screen and (max-width: 768px)",
        lg: "screen and (max-width: 1024px)",
        xl: "screen and (max-width: 1280px)",
    },
};
