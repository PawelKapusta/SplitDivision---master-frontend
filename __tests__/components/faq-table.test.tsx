import React from "react";
import { render } from "@testing-library/react";
import FAQ from "@components/faq-table/index";
import { mockTheme } from "../mocks/theme";
import { ThemeProvider } from "styled-components";
import { useSelector } from "react-redux";

jest.mock("react-redux", () => ({ useSelector: jest.fn() }));
describe("faq-table component", () => {
    beforeEach(() => {
        (useSelector as jest.Mock).mockReturnValue({
            faqs: [
                { id: 1, question: "Q1", answer: "A1" },
                {
                    id: 2,
                    question: "Q2",
                    answer: "A2",
                },
            ],
        });
    });
    it("renders correctly", () => {
        const { container } = render(
            <ThemeProvider theme={mockTheme}>
                <FAQ />
            </ThemeProvider>,
        );
        expect(container).toMatchSnapshot();
    });
});
