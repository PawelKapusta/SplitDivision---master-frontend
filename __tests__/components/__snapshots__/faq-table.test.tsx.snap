// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`faq-table component renders correctly 1`] = `
<div>
  <div
    class="sc-b0e1a03d-0 hfhUIA"
  >
    <h2>
      screens.faq.title
    </h2>
    <ul
      class="sc-b0e1a03d-1 dwLlxZ"
    >
      <li>
        <h3
          class="sc-b0e1a03d-2 hYLNXS faq-question"
        >
          Q1
        </h3>
        <div
          class="sc-b0e1a03d-3 khYBjR faq-answer"
        >
          A1
        </div>
      </li>
      <li>
        <h3
          class="sc-b0e1a03d-2 hYLNXS faq-question"
        >
          Q2
        </h3>
        <div
          class="sc-b0e1a03d-3 khYBjR faq-answer"
        >
          A2
        </div>
      </li>
    </ul>
  </div>
</div>
`;
